MaterialLib/MPL/Component.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \date   Sep 7, 2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+#pragma once
+
+#include "Property.h"
+
+namespace MaterialPropertyLib
+{
+/// \brief This class defines components (substances).
+/// \details The Component class is a base class used for not further specified
+/// components. Components are specified by the property 'name'. For specified
+/// components we derive special classes from this class (for clarity they are
+/// located in the 'components' subfolder).
+class Component
+{
+public:
+    /// Default constructor of Component. This constructor is used
+    /// when the component is not specified via the 'name'-tag.
+    Component();
+
+    /// Constructor for a custom component
+    Component(std::string const& component_name,
+              std::unique_ptr<PropertyArray>&& properties);
+    virtual ~Component() = default;
+
+    /// A get-function for retrieving a cartain property.
+    Property const& property(PropertyType const& /*p*/) const;
+
+    template <typename T>
+    T value(PropertyType const p) const
+    {
+        return property(p).template value<T>();
+    }
+
+    template <typename T>
```

Comment:
Is my understanding correct: the variable_array is used to pass information for property value calculation?

Reply:
Yes.